### PR TITLE
Add com.arviceblot.eso-addon-manager

### PR DESCRIPTION
**Pull request by arviceblot**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** [https://github.com/arviceblot/eso-addons](https://github.com/arviceblot/eso-addons)
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by arviceblot** [comment]
bot, build com.arviceblot.eso-addon-manager

File: com.arviceblot.eso-addon-manager.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{
+    "app-id": "com.arviceblot.eso-addon-manager",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "22.08",
```
Comment (hfiguiere) [inline com.arviceblot.eso-addon-manager.json:4]: use 23.08

File: com.arviceblot.eso-addon-manager.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command": "eso-addon-manager",
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=dri",
+        "--filesystem=host"
```
Comment (hfiguiere) [inline com.arviceblot.eso-addon-manager.json:16]: `host` is a very broad filesystem permission

File: com.arviceblot.eso-addon-manager.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                "env": {
+                    "CARGO_HOME": "/run/build/eso-addon-manager/cargo"
+                }
+            },
+            "build-commands": [
+                "cargo --offline fetch --manifest-path Cargo.toml --verbose",
+                "cargo --offline build --release --verbose",
+                "install -Dm755 ./target/release/eso-addon-manager -t /app/bin/",
+                "install -Dm644 ./data/${FLATPAK_ID}.desktop -t /app/share/applications/",
+                "install -Dm644 ./data/${FLATPAK_ID}.svg -t /app/share/icons/hicolor/scalable/apps/"
```
Comment (hfiguiere) [inline com.arviceblot.eso-addon-manager.json:35]: I don't see the required Appstream file being installed.

File: com.arviceblot.eso-addon-manager.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
+{
+    "app-id": "com.arviceblot.eso-addon-manager",
```
Comment (hfiguiere) [inline com.arviceblot.eso-addon-manager.json:2]: `arviceblot.com` doesn't seem to indicate it's linked to this package. I get an error page.

File: com.arviceblot.eso-addon-manager.json (hunk 2 above)
Comment (arviceblot) [reply com.arviceblot.eso-addon-manager.json:16]: This is for various source game install location options. They could be installed anywhere, but directory is selected by user. The original intent is to run on Steam Deck where games can be moved from SSD to SD Card, etc.

File: com.arviceblot.eso-addon-manager.json (hunk 3 above)
Comment (arviceblot) [reply com.arviceblot.eso-addon-manager.json:35]: It was unclear to me from the submission requirements whether the Appstream file was required. I'll work on putting one together.

File: com.arviceblot.eso-addon-manager.json (hunk 4 above)
Comment (arviceblot) [reply com.arviceblot.eso-addon-manager.json:2]: It is my personal domain, there's no real web content for this project outside the github repo. I have added a TXT entry `eso-addon-manager.arviceblot.com` for some proof.

File: com.arviceblot.eso-addon-manager.json (hunk 3 above)
Comment (hfiguiere) [reply com.arviceblot.eso-addon-manager.json:35]: it's listed on the requirement page....

Comment (arviceblot) [reply com.arviceblot.eso-addon-manager.json:35]: Yes, under the [Acceptible](https://docs.flathub.org/docs/for-app-authors/requirements#acceptable-but-should-be-submitted-upstream) section. If it is required it should be listed under [Required](https://docs.flathub.org/docs/for-app-authors/requirements#required-files). I'll work on making one anyways.

File: com.arviceblot.eso-addon-manager.json (hunk 1 above)
Comment (arviceblot) [reply com.arviceblot.eso-addon-manager.json]: Updated.

**Comment by arviceblot** [comment]
bot, build com.arviceblot.eso-addon-manager

File: com.arviceblot.eso-addon-manager.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+                    "CARGO_HOME": "/run/build/eso-addon-manager/cargo"
+                }
+            },
+            "build-commands": [
+                "cargo --offline fetch --manifest-path Cargo.toml --verbose",
+                "cargo --offline build --release --verbose",
+                "install -Dm755 ./target/release/eso-addon-manager -t /app/bin/",
+                "install -Dm644 ./data/${FLATPAK_ID}.desktop -t /app/share/applications/",
+                "install -Dm644 ./data/${FLATPAK_ID}.svg -t /app/share/icons/hicolor/scalable/apps/",
+                "install -Dm644 ./data/com.arviceblot.eso_addon_manager.metainfo.xml -t /app/share/metainfo"
```
Comment (hfiguiere) [inline com.arviceblot.eso-addon-manager.json:36]: it's the wrong name.

Comment (hfiguiere) [reply com.arviceblot.eso-addon-manager.json:36]: and the app-id in it is incorrect too.

Comment (hfiguiere) [reply com.arviceblot.eso-addon-manager.json:36]: also it won't validate. missing required `releases` and `content_rating`.

**Comment by arviceblot** [comment]
bot, build com.arviceblot.eso-addon-manager

**Comment by arviceblot** [comment]
bot, build com.arviceblot.eso-addon-manager

File: com.arviceblot.eso-addon-manager.json (hunk 5 above)
Comment (arviceblot) [reply com.arviceblot.eso-addon-manager.json]: updated

**Comment by arviceblot** [comment]
> I only see a rotating spinner here

Thank you for testing. I have pushed an upstream fix for empty config, will build shortly

**Comment by arviceblot** [comment]
bot, build com.arviceblot.eso-addon-manager

**Pull request closed without merging**